### DISCUSSION TITLE
Schema detection and evolution. Looking for alternatives to `snowflake_unsafe_execute` resource due to the deprecation notice

### DISCUSSION THREAD

**Discussion by Arsenikki** [Snowflake-Labs/terraform-provider-snowflake#2698] [Q&A]
Hey,

I'm looking for some ideas as my company is rapidly scaling the amount of environments we need to manage with differing deployment lifecycles. This comes with the challenge that the application database schemas might differ from env to env in case different versions of our stack is running. To keep it all somewhat manageable, I've been experimenting with Snowflake's schema detection and schema evolution features to help keep Snowflake DBs in sync with our application side

Unfortunately neither Schema Evolution or Schema Detection seem to be natively supported by this Terraform provider, so I've been looking for workarounds. I found [snowflake_unsafe_execute](https://registry.terraform.io/providers/Snowflake-Labs/snowflake/latest/docs/resources/unsafe_execute) to handle these both and with my initial tests this has worked well enough. However, there's now a deprecation notice set for this resource:

```Experimental resource. Will be deleted in the upcoming versions. Use at your own risk.```

The deprecation notice doesn't suggest any alternatives, but are there some other ways of doing this through Terraform that I'm missing?

**Comment by sfc-gh-asawicki** [comment]
Hey @Arsenikki. Thanks for reaching out to us.

I gave an answer to a similar question today here: https://github.com/Snowflake-Labs/terraform-provider-snowflake/issues/1600#issuecomment-2048964890. The bottom line is, that we won't remove it without providing an alternative.
